support/scripts/gcov/gcov_process.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+# Licensed under the MIT License (the "License", see COPYING.md).
+# You may not use this file except in compliance with the License.
+
+set -e
+
+if [[ $# -ne 3 ]]; then
+	echo "Usage: $0 [-b <binary_dump> | -c <console_text>] <build_dir>" 1>&2
+	exit 1
+fi
+
+if [ -n "${CROSS_COMPILE}" ]; then
+	GCOV=${CROSS_COMPILE}/gcov
+else
+	GCOV=$(which gcov)
+fi
+
+if [ -z "$GCOV" ]; then
+	echo "$0: Could not find gcov" 1>&2
+	exit 1
+fi
+
+SCRIPT_DIR=$(realpath $(dirname "$0"))
+CONVERTED_FILE=$(realpath dump.mem)
+PARSE_SCRIPT=$(realpath "$SCRIPT_DIR"/gcov_serial_parse.py)
+OUTPUT_FILE=$(realpath "$2")
+BUILD_DIR=$(realpath "$3")
+
+cd "$SCRIPT_DIR"
+
+mkdir -p objs
+mkdir -p results
+
+rm -rf objs/*
+rm -rf results/*
+
+find "$BUILD_DIR" -type d -name objs -prune -o -name \*.gcno -exec cp {} objs/ \;
+
+lcov --gcov-tool "$CROSS_COMPILE"gcov --capture --initial \
+     --directory objs/ -o results/baseline.info
+
+if [[ $1 == "-c" ]]; then
+	"$PARSE_SCRIPT" --filename "$OUTPUT_FILE" --output "$CONVERTED_FILE"
+	gcov-tool merge-stream < "$CONVERTED_FILE"
+else
+	gcov-tool merge-stream < "$OUTPUT_FILE"
+fi
+
+find "$BUILD_DIR" -type d -name objs -prune -o -name \*.gcda -exec cp {} objs/ \;
+
+echo
+
+./lcov_newcoverage.sh
```

Comment:
Looking at the scripts it seems that we have a bit too many. The ones we import from `embedded-gcov` especially do some parameter validation to handle cases of per-test runs and invoke a single command. Perhaps we can simplify this to the following and remove the scripts altogether.
```
-./lcov_newcoverage.sh
-./lcov_combine_new_base.sh
-./genhtml_report.sh
+# Capture converage info from this run
+lcov --gcov-tool gcov --capture --directory objs/ \
+       --ignore-errors gcov,source -o results/run.info
+
+# Combine baseline and coverage info from this run
+lcov -a results/base.info -a results/uk.info -o results/run.info
+
+# Generate HTML coverage report
+genhtml -o results/html results/total.info --show-details --legend
```
That would leave us with two scripts: `gcov_process.sh`, and `gcov_serial_parse.py`. So maybe it would be even better if we provided a single python script of our own, for all processing.

PS: This would also require updating `README.md`.